src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -0,0 +1,233 @@
+)
+
+
+class ObjectsAPIRegistrationHandler(Protocol[OptionsT]):
+    """Provide the registration data to be sent to the Objects API."""
+
+    def get_object_data(
+        self,
+        submission: Submission,
+        options: OptionsT,
+    ) -> dict[str, Any]:
+        """Get the object data payload to be sent to the Objects API."""
+        ...
+
+    def get_update_payment_status_data(
+        self, submission: Submission, options: OptionsT
+    ) -> dict[str, Any]: ...
+
+
+class ObjectsAPIV1Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV1]):
+    """Provide the registration data for legacy (v1) registration options, using JSON templates."""
+
+    @staticmethod
+    def _point_coordinate(value):
+        if not value or not isinstance(value, list) or len(value) != 2:
+            return SKIP
+        return {"type": "Point", "coordinates": [value[0], value[1]]}
+
+    @staticmethod
+    def get_payment_context_data(submission: Submission) -> dict[str, Any]:
+        return {
+            "completed": submission.payment_user_has_paid,
+            "amount": str(submission.payments.sum_amount()),
+            "public_order_ids": submission.payments.get_completed_public_order_ids(),
+        }
+
+    @staticmethod
+    def build_options(plugin_options: RegistrationOptionsV1, key_mapping: dict) -> dict:
+        """
+        Construct options from plugin options dict, allowing renaming of keys
+        """
+        options = {
+            new_key: plugin_options[key_in_opts]
+            for new_key, key_in_opts in key_mapping.items()
+            if key_in_opts in plugin_options
+        }
+        return options
+
+    def register_submission_csv(
+        self,
+        submission: Submission,
+        options: RegistrationOptionsV1,
+        documents_client: DocumentenClient,
+    ) -> str:
+        if not options.get("upload_submission_csv", False):
+            return ""
+
+        if not options["informatieobjecttype_submission_csv"]:
+            return ""
+
+        submission_csv_options = self.build_options(
+            options,
+            {
+                "informatieobjecttype": "informatieobjecttype_submission_csv",
+                "organisatie_rsin": "organisatie_rsin",
+                "doc_vertrouwelijkheidaanduiding": "doc_vertrouwelijkheidaanduiding",
+                "auteur": "auteur",
+            },
+        )
+        qs = Submission.objects.filter(pk=submission.pk).select_related("auth_info")
+        submission_csv = create_submission_export(qs).export("csv")  # type: ignore
+
+        submission_csv_document = create_csv_document(
+            client=documents_client,
+            name=f"{submission.form.admin_name} (csv)",
+            csv_data=submission_csv,
+            options=submission_csv_options,
+            language=submission.language_code,
+        )
+
+        return submission_csv_document["url"]
+
+    @override
```

Comment:
I added a comment [here](https://github.com/open-formulieren/open-forms/pull/3954#discussion_r1505994269) related to this. This decorator was first introduced for type checkers, with the canonical example being:

```python
class Base:
    def base_method(self): ...

class Sub(Base):
    def base_method(self): ...
```

In this scenario, you could change `Base.base_method` to be `Base.new_method`, and type checkers wouldn't be aware of such a change (they would see `Sub.base_method` as a new method not present on `Base`). By adding the `@override` decorator on `Sub.base_method`, it will require the method to exist in `Base`, essentially catching the issue I described.

Because we don't run type checking, it doesn't really change anything in our case, but I find it easier to see which method is tied to the plugin ABC, especially with our plugin architecture.

> Are we going to keep it and use it generally?

I guess it's personal preference